packages/wonder-stuff-sentry/src/collate-sentry-data.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import {
+    errorsFromError,
+    Order,
+    getKindFromError,
+} from "@khanacademy/wonder-stuff-core";
+import {EmptySentryData} from "./empty-sentry-data.js";
+import {sentryDataReducer} from "./sentry-data-reducer.js";
+import {getOptions} from "./init.js";
+import {KindSentryError} from "./kind-sentry-error.js";
+import type {SentryData} from "./types.js";
+
+/**
+ * Collate sentry data from a given error and its causal errors into one object.
+ *
+ * Parses the error and all causal errors, merging any SentryData found to
+ * produce a single SentryData object.
+ *
+ * For `tags` and `contexts`, existing values of inner (earlier) errors are
+ * overwritten by outer (later) errors. For `fingerprints`, the arrays are
+ * concatenated together.
+ *
+ * Each causal error is added as a context containing its original stack,
+ * sentry data, and message.
+ *
+ * Finally, tags for the error kind, grouping, and message concatenation are
+ * added.
+ *
+ * @param {Error} error The error to be collated into sentry data.
+ * @returns {$ReadOnly<SentryData>} A single SentryData object combining
+ * all the SentryData from this and causal errors it contains.
+ */
+export const collateSentryData = (error: Error): $ReadOnly<SentryData> => {
+    // First, get all the errors in the stack.
+    const errors = errorsFromError(error, Order.ConsequenceFirst);
+    const consquenceAndCauses: $ReadOnlyArray<Error> = Array.from(errors);
```

Comment:
I take back my earlier comment about you having a spell-checker, loooool... you just must have great spelling skills, since there are almost 0 spelling errors! But found one!